src/OptimalTransport.jl

```diff
@@ -125,26 +131,36 @@ The Gibbs kernel `K` is given by `K = exp.(- C / eps)` where `C` is the cost mat
 `eps` the entropic regularization parameter. The optimal transport plan for histograms `u`
 and `v` and cost matrix `C` with regularization parameter `eps` can be computed as
 `Diagonal(u) * K * Diagonal(v)`.
+
+Note that for a common kernel `K`, multiple histograms may be provided for a batch computation by passing `mu` and `nu`
+as matrices whose columns `mu[:, i]` and `nu[:, i]` correspond to pairs of histograms. 
+The output are then matrices `u` and `v` such that `u[:, i]` and `v[:, i]` are the dual variables for `mu[:, i]` and `nu[:, i]`.
 """
 function sinkhorn_gibbs(mu, nu, K; tol=1e-9, check_marginal_step=10, maxiter=1000)
-    if !(sum(mu) ≈ sum(nu))
+    if !(sum(mu; dims=1) ≈ sum(nu; dims=1))
         throw(ArgumentError("Error: mu and nu must lie in the simplex"))
     end
-
+    if size(mu, 2) != size(nu, 2)
```

Comment:
I think we might also want to allow e.g. `size(mu, 2) == 1` (this is the case even if `mu` is a vector) and `size(nu, 2) > 1` and vice versa. The main point just seems to be to ensure that `u` and `v` are sufficiently "big". I assume that one could maybe avoid adding a custom implementation here and instead use some functionality in base Julia that deals with shapes in broadcasting operations. I'll check this later today.

Reply:
Ok I've implemented one way of doing this by broadcasting. I'll try and rebase this since the code's been updated in the meantime.